### PR TITLE
Fixed issue #12745

### LINKED ISSUE #12745: Copying code from the website also copies the prompts
## Description
When copying code using the Copy from Clipboard javascript on the website, it also is copying the prompt
Go to 1) http://mxnet.incubator.apache.org/install/index.htmlplatform=Linux&language=Python&processor=GPU
2) use the copy to Clipboard option to copy text it has copied,
`$ pip install mxnet-cu92` earlier the shell prompt was not copied to clipboard

## Environment info (Required): 
Mac OS/Google Chrome Version 69.0.3497.100 (Official Build) (64-bit)

@aaronmarkham could you please take a look

### PR DESCRIPTION
**Pull request by LuckyPigeon**
## Description ##
Filter prompt sign while copying code from the website. Fixed #12745 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by aaronmarkham** [comment]
I checked http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-12781/2/tutorials/gluon/mnist.html
And I found that the first code block will still copy the $.

File: docs/_static/js/copycode.js [hunk 1]
```diff
@@ -62,6 +62,13 @@ $(document).ready(function(){
...
 
         for(var i = 0; i < lines.length; ++i) {
+            for(var lang in LANG_GP) {
+                if(lines[i].indexOf(LANG_GP[lang]) != -1) {
+                    var re = new RegExp(LANG_GP[lang], g);
+                    lines[i] = lines[i].replace(re, "");
+                    break;
+                }
+            }
             lines[i] = lines[i].replace(/^\s+|\s+$/g, "");
```
Comment (aaronmarkham) [inline docs/_static/js/copycode.js]: I know you didn't change this line, but it seems to be throwing errors in the preview.
```
Uncaught ReferenceError: g is not defined
    at copycode.js:67
    at t.emit (clipboard.min.js:7)
    at e (clipboard.min.js:7)
    at e (clipboard.min.js:7)
    at e (clipboard.min.js:7)
    at e (clipboard.min.js:7)
    at new e (clipboard.min.js:7)
    at t.e (clipboard.min.js:7)
    at HTMLBodyElement.<anonymous> (clipboard.min.js:7)
    at HTMLBodyElement.<anonymous> (clipboard.min.js:7)
```
There's also some 403 errors happening which really are 404's.  Seems like there might be a bigger issue going on here...

Comment (aaronmarkham) [reply docs/_static/js/copycode.js]: Actually, it is a line in the updated code block... line 67...
```
var re = new RegExp(LANG_GP[lang], g);
```

Comment (LuckyPigeon) [reply docs/_static/js/copycode.js]: Fixed.

**Comment by aaronmarkham** [comment]
Looks like we're hung up on a flaky test. I opened an issue here: https://github.com/apache/incubator-mxnet/issues/12797
I'll retrigger CI, but it might just fail again until that test is disabled.

**Comment by aaronmarkham** [comment]
Now tracking https://github.com/apache/incubator-mxnet/pull/12788 - might need to rebase after this PR gets merged.

**Comment by LuckyPigeon** [comment]
@aaronmarkham I find that #12788 has closed, but I can't find the new PR issue related with it. Does the changes are already Updated?

**Comment by sandeep-krishnamurthy** [comment]
@LuckyPigeon - The regression is reverted and master is Green now. Can you please rebase your branch once?

**Comment by aaronmarkham** [comment]
@LuckyPigeon Let me know if you need any help.

**Comment by LuckyPigeon** [comment]
@aalexandrov This is my first time using git rebase, this shows up after I rebase.
```
First, rewinding head to replay your work on top of it...
Applying: Fixed issue 12745
Applying: Fix copycode.js error caused by new code block
```
Is this means rebase successfully?

**Comment by aaronmarkham** [comment]
> @aalexandrov This is my first time using git rebase, this shows up after I rebase.
> 
> ```
> First, rewinding head to replay your work on top of it...
> Applying: Fixed issue 12745
> Applying: Fix copycode.js error caused by new code block
> ```
> Is this means rebase successfully?
> Did I need to push after I rebase?

It looks like it worked, but I think you still need to `git push` now. We should see another commit, and that commit will trigger CI.

**Comment by LuckyPigeon** [comment]
I encounter a problem while I using `git push`
image

**Comment by aaronmarkham** [comment]
Take a look at https://cwiki.apache.org/confluence/display/MXNET/MXNet+Development+Guide

The section that talks about making a PR has this as a guide:

```
git remote add upstream https://github.com/apache/incubator-mxnet
git fetch upstream
git rebase upstream/master
```
After going through the rebase process, you might have to force push with:
```
git push --force
```

**Comment by LuckyPigeon** [comment]
@aaronmarkham After I go through those commands, I encounter this error.
image

**Comment by aaronmarkham** [comment]
You can't push upstream.
Just do 'git push --force'
That will force push to your fork and it will automatically update your pull request.

**Comment by LuckyPigeon** [comment]
@aaronmarkham  It's seems I can't do git push --force, after I do the command and it shows this message.
image

**Comment by aaronmarkham** [comment]
Maybe try the suggested command and then force? I'm not sure how you're setup. If you'd like we can get on slack and troubleshoot in realtime. I'm in California. Let me know a time that works for you (I'm typically available 845am - 615pm).

**Comment by aaronmarkham** [comment]
Preview: http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-12781/6/index.html

Unfortunately, what I'm seeing on the install page, is that it is still copying the `$`.
http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-12781/6/install/index.html?platform=Linux&language=Python&processor=CPU

**Comment by LuckyPigeon** [comment]
@aaronmarkham I still don't get it. Why the install page still copying the bash prompt. I have test it functionally on my local machine, but when I put it on the web, it couldn't work anymore.

**Comment by stu1130** [comment]
@aaronmarkham ping again

**Comment by aaronmarkham** [comment]
There are about a half dozen `document.ready` functions that trigger, so this could be an issue with the js.
The fact that it is broken in the CI preview means that you can emulate it locally when you build the docs (running `make html` from the docs folder). 

I was able to see the issue on another page, so it isn't limited to the install page:
http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-12781/6/tutorials/gluon/mnist.html

@LuckyPigeon can you see that it isn't working on the above link? Can you debug it from there to see why your change isn't working?

**Comment by vandanavk** [comment]
@mxnet-label-bot update [pr-awaiting-response]

**Comment by marcoabreu** [comment]
@LuckyPigeon there are open comments from @aaronmarkham , please address

**Comment by LuckyPigeon** [comment]
@marcoabreu thanks!

**Comment by sandeep-krishnamurthy** [comment]
@LuckyPigeon - Did you get a chance to update this PR? This is very useful for users.

**Comment by LuckyPigeon** [comment]
@sandeep-krishnamurthy Recently, I have started my vacation, and I can continue this PR again.

**Comment by sandeep-krishnamurthy** [comment]
@LuckyPigeon - When you update, please let me know, we can take this to completion.

**Comment by LuckyPigeon** [comment]
@sandeep-krishnamurthy Updated!

**Comment by LuckyPigeon** [comment]
@aaronmarkham Does the preview link build successfully? Because I build the code locally and shows the right result.
 Here's my local doc build:
screenshot from 2019-02-10 10-42-23

**Comment by aaronmarkham** [comment]
No, the preview link I gave in my last comment shows that it isn't working. The $ is still copied.

**Comment by LuckyPigeon** [comment]
@aaronmarkham I'm not sure what went wrong while building docs. I've tried built the code of this pull request on ubuntu 1804 and the `1` sign can be removed successfully. So that must be some other thing went wrong while building the docs. Can you recommend some people for help solving the problem? Thanks!

**Comment by aaronmarkham** [comment]
Can you share a preview link to where you have it working? Maybe I can see what's different.

**Comment by LuckyPigeon** [comment]
But I build it on my local machine, so I don't have a preview link. I can only share the `_build` folder after I built the docs successfully.

**Comment by anirudhacharya** [comment]
@LuckyPigeon @aaronmarkham please revisit this PR and see it to completion.

**Comment by karan6181** [comment]
@LuckyPigeon Can you please address the review comments?

@aaronmarkham Could you please review this PR again?

Thank you!

**Comment by LuckyPigeon** [comment]
@karan6181 Sure, but I not quite understand what went wrong in the code. Because the amazon machine can't filter the prompt with the code, but when I build the docs on the local machine success.

**Comment by piyushghai** [comment]
@LuckyPigeon Were you able to figure out the issue blocking this PR ?

**Comment by Roshrini** [comment]
Thanks for working on this. This will be useful for uers. @aaronmarkham Can you help @LuckyPigeon by any chance?

**Comment by aaronmarkham** [comment]
The preview link doesn't work anymore. I'm not convinced this is fixed. Seems like a larger issue, really.

**Comment by roywei** [comment]
@LuckyPigeon Thanks for the contribution, could you address the comments?

**Comment by pinaraws** [comment]
@LuckyPigeon Thanks for the contribution, could you address the comments?

**Comment by LuckyPigeon** [comment]
@roywei @pinaraws Sure. But the problem seems has nothing to do with the code of `copycode.js` , I think it's something wrong while building the docs, I have encounter it recently, but I am still not sure which part goes wrong.

**Comment by piyushghai** [comment]
@LuckyPigeon @aaronmarkham  What's the path forward on this PR ?

**Comment by Roshrini** [comment]
@aaronmarkham Is there any issue with docs build as @LuckyPigeon Mentioned?

**Comment by aaronmarkham** [comment]
It is known that the website publish process (including python module `beautiful soup` doing code injections, along with some of the site's javascript, cause or have bugs. These make the DOM and interactions therewith unpredictable. The full production build to recreate such issues is bulky, difficult, as well as resource and time consuming. I fear that without front-end engineering help and resources (access to sufficient build/server allocations) @LuckyPigeon is going to be forever stuck on this PR. I have looked at the issue, and I see it as a rabbit hole of javascript and publishing bugs that need to be fixed.

**Comment by karan6181** [comment]
@LuckyPigeon Ping for the update? How should we drive this PR to resolution? Thanks!

File: docs/_static/js/copycode.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/*Copy code to clipboard*/
```
Comment (szha) [inline docs/_static/js/copycode.js]:
```suggestion
```

**Comment by piyushghai** [comment]
@LuckyPigeon Gentle ping for drive this PR forward.

**Comment by LuckyPigeon** [comment]
@Roshrini @aaronmarkham @karan6181 @piyushghai
Sorry for late reply. I have followed the [mxnet doc build](https://cwiki.apache.org/confluence/display/MXNET/How+to+Build+the+Website) instruction and build a doc on my local computer before, and sometimes it will encounter a problem that after I update my code and execute `$ make html`, it can't detect the update, and after executed the command, `index.html` stays the same as before.
Like @aaronmarkham said before, I think I am going to need front-end engineering help, who develop the `mxnet docs` for the best, and I am going to trace the doc code and find out where the problem is.

**Comment by aaronmarkham** [comment]
Closing since the new site doesn't have this function/issue.

**Pull request closed without merging**